misc/python/materialize/test_analytics/setup/views/02-build-job.sql

```diff
@@ -150,10 +150,18 @@ WHERE d.predecessor_index <> 0
 ;
 
 CREATE OR REPLACE MATERIALIZED VIEW mv_recent_build_job_success_on_main IN CLUSTER test_analytics AS
-SELECT *
+SELECT
+    build_job_id,
+    predecessor_build_number,
+    predecessor_build_id,
+    predecessor_build_job_id,
+    predecessor_build_step_success,
+    predecessor_index,
+    predecessor_is_latest_retry
 FROM v_build_job_success
 WHERE branch = 'main'
-AND date + INTERVAL '30' DAY > mz_now();
+AND date + INTERVAL '30' DAY > mz_now()
+AND predecessor_index <= 10;
```

Comment:
I'm wondering if Mz can understand how the index works or still keeps all objects in memory.

Reply:
I guess the question is whether the filter gets pushed down? I hope so but am not 100 % sure.

Reply:
I can additionally decrease the retention period to a shorter interval.

Reply:
See for example https://materialize.com/blog/power-user/#memory-footprint-of-queries In Materialize some operators like min/max can keep only one element in memory if deletions are not possible (append-only sources). But since we are using tables here a min/max would require lots of memory to keep all results in memory, since the current min/max value could be deleted at any point.

Reply:
If we used an append-only source (like a webhook), we could probably reduce memory by a lot for something like "get the last 10 entries"

Reply:
I don't think we are using any too bad operators (at least no min / max) -- besides heavy use of WMR. But, yeah, I understand that append-only source might still bring some benefits.

Please also be aware that this MV will not contains 10 builds; it will contain the last 10 executions of all build jobs.